cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-core/src/test/java/com/sap/cloud/cf/monitoring/client/configuration/EnvUtils.java

```diff
@@ -0,0 +1,32 @@
+package com.sap.cloud.cf.monitoring.client.configuration;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class EnvUtils {
+
+    public static void setEnvs(String[][] envs) throws Exception {
+        Map<String, String> envMap = Stream.of(envs).collect(Collectors.toMap(data -> data[0], data -> data[1]));
+
+        setNewEnvs(envMap);
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private static void setNewEnvs(Map<String, String> newenv) throws Exception {
```

Comment:
Currently we are using java 8 where jigsaw is no available. The code is used in tests. Oneday when we migrate to the next version of java we can refactor that.

Reply:
Users of the library may use it with higher Java versions. Please test, that they do not run into problems, when using the library. It would be a poor choice to cut of support for current Java versions.

Reply:
I just checked, that there are several issues with newer Java versions in the logging library. We will need a bigger effort to address those. This part actually compiles fine with SapMachine 12.0.1. The tests are also green.